src/wwwroot/controllers/LoginCtrl.js

```diff
@@ -5,65 +5,95 @@
+  function LoginCtrl($location, $timeout, auth, $translate, vcRecaptchaService) {
 
-    $scope.forgotOpened = false;
-    $scope.forgotSuccessful = false;
+    var vm = this;
+    vm.forgotOpened = false;
+    vm.forgotSuccessful = false;
+    vm.updateValidation = updateValidation;
+    vm.gotoBottom = gotoBottom;
+    vm.submitLogin = submitLogin;
+    vm.removeSucceed = removeSucceed;
+    vm.submitForgot = submitForgot;
 
-    $scope.updateValidation = function () {
-      if ($scope.loginform && $scope.loginform.email) {
-        $scope.loginform.email.$setValidity('error', true);
+    function updateValidation(loginform) {
+      if (loginform && loginform.email) {
+        loginform.email.$setValidity('error', true);
       }
-      if ($scope.loginform && $scope.loginform.password) {
-        $scope.loginform.password.$setValidity('error', true);
+      if (loginform && loginform.password) {
+        loginform.password.$setValidity('error', true);
       }
     }
-    $scope.gotoBottom = function () {
-      $scope.forgotOpened = !$scope.forgotOpened;
+    function gotoBottom() {
+      vm.forgotOpened = !vm.forgotOpened;
     };
 
-    $scope.submitLogin = function () {
-      if ($scope.loginform.$invalid) {
+    function submitLogin (loginform) {
+      if (loginform.$invalid) {
         return;
       }
 
       var credentials = {
-        username: $scope.email,
-        password: $scope.password
+        username: loginform.email.$modelValue,
+        password: loginform.password.$modelValue
       };
 
       auth.login(credentials).then(function (result) {
         if (result.authenticated) {
           $location.path('/');
         } else {
-          $scope.loginform.email.$setValidity('error', false);
-          $scope.loginform.password.$setValidity('error', false);
+          loginform.email.$setValidity('error', false);
+          loginform.password.$setValidity('error', false);
         }
       });
     };
 
-    $scope.removeSucceed = function () {
-      if ($scope.forgotSuccessful) {
-        $scope.forgotSuccessful = false;
+    function removeSucceed(forgotEmail) {
+      if (vm.forgotSuccessful) {
+        vm.forgotSuccessful = false;
       }
     }
 
-    $scope.submitForgot = function () {
-      if ($scope.forgotForm.$invalid) {
+    function submitForgot(forgotForm) {      
+      if (forgotForm.$invalid) {
         return;
       }
+      vcRecaptchaService.execute(vm.widgetId);
+    };
+
+    vm.setResponse = setResponse;
+    vm.setWidgetId = setWidgetId;
+    vm.reloadCaptcha = reloadCaptcha;
 
-      auth.forgotPassword($scope.forgotEmail, $translate.use())
-        .then(function () {
-            $scope.forgotSuccessful = true;
-        })
+    function setWidgetId (widgetId) {
+        vm.widgetId = widgetId;
     };
+    
+    function reloadCaptcha () {
+        vcRecaptchaService.reload(vm.widgetId);
+        vm.response = null;
+    };
+
+    function setResponse (response) {
```

Comment:
Maybe _setCaptchaResponse_ could be helpful?

I would like to make more explicit the flow, right now it is difficult to understand how events are being executed.

I think that it is: `submitForgot -> setResponse -> forgotPassword -> reloadCaptcha` but it is not so clear. I am not sure if it is possible to represent it in a better way.